db_test.go

```diff
@@ -437,3 +438,57 @@ func assertContains[K fmt.Stringer](s *SurrealDBTestSuite, input []K, matcher fu
 	s.NotEmptyf(matching, "Input %+v did not contain matching element", fmt.Sprintf("%+v", input))
 	return matching
 }
+
+func TestIsDuplicateUniqueIdx(t *testing.T) {
```

Comment:
It can be better if we use Testify's Suit and test with the database itself for testing unexpected behavior.